src/main/java/com/cs56fitnessapp/models/Meal.java

```diff
@@ -1,29 +1,137 @@
 package com.cs56fitnessapp.models;
+import java.util.ArrayList;
+
 
 /**
- * @author Nerissa Hsieh
+ * @author Nerissa Hsieh & Jimwell Castillo
  * Date Created: 10/03/17
  * Date Modified: 10/03/17
  */
 
-public class Meal implements DietFacts {
-    @Override
-    public int getCaloriesIn() {
-        return 0;
-    }
+public class Meal implements DietFacts
+    {
+        @Override
+        public int getCaloriesIn() {
+            return 0;
+        }
 
-    @Override
-    public double getTotalFatIn() {
-        return 0;
-    }
+        @Override
+        public double getTotalFatIn() {
+            return 0;
+        }
+
+        @Override
+        public double getTotalCarbIn() {
+            return 0;
+        }
+
+        @Override
+        public double getProteinIn() {
+            return 0;
+        }
+
+
+        /** Edited Jimwell Castillo
+         * 10.18.17
+         * */
+
+        public class Meals
+            {
+
+                private String mealName;
+                private ArrayList <FoodEntry> meal;
+
+                public Meals (String mealName )
+                    {
+                        this.mealName = mealName;
+                        this.meal= new ArrayList<>();
+                    }
+
+                /**Checks to see if
+                 food entry has been entered
+                */
+
+                public boolean addNewMeal(FoodEntry food)
+                    {
+                        if (findName(food.getFoodName()) >= 0)
+                        {
+                            System.out.println("Food is already Entered");
```

Comment:
When we're implementing model classes like `User`, `Workout`, `Meal` - we have to implement them in a pure way. It means that I give you some data and your class API returns me data back without producing side effects. 
 System.out.println("Food is already Entered"); is a side effect. 
Note, when, you're running ItelliJ application (which is Java application) it does not print anything to your terminal console.